pkg/admission/zone_tracker.go

```diff
@@ -0,0 +1,231 @@
+package admission
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/rollout-operator/pkg/config"
+	"github.com/thanos-io/objstore"
+	"github.com/thanos-io/objstore/providers/azure"
+	"github.com/thanos-io/objstore/providers/gcs"
+	"github.com/thanos-io/objstore/providers/s3"
+	"gopkg.in/yaml.v2"
+	appsv1 "k8s.io/api/apps/v1"
+)
+
+type zoneTracker struct {
+	mu    sync.Mutex
+	zones map[string]string
```

Comment:
Should we store some struct inside the map rather than just a string? With just the string, this is more of a lastDownscaledTracker than a zone tracker, and I'm wondering if we may want to store other things in there in the future, storing a separate JSON object per zone may make migration easier.

Reply:
This seems like a reasonable improvement and hopefully also resolves the issue mentioned above with `stsList` only applying to a specific rollout group?

Reply:
Any suggestions for what that struct might look like?

Reply:
Just something like:

```go
type zoneInfo struct {
	lastDownscaled time.Time
}
```

Reply:
I implemented this! It's working on initial testing. Thanks for the suggestion.